cartridges/int_adyen_controllers/cartridge/controllers/Adyen.js

```diff
@@ -370,12 +370,20 @@ function cancelOrRefund() {
  */
 function authorizeWithForm()
 {
-	var	adyen3DVerification = require('int_adyen_overlay/cartridge/scripts/adyen3DVerification'), result,
-	order = session.custom.order,
-	paymentInstrument = session.custom.paymentInstrument,
+	var	adyen3DVerification = require('int_adyen_overlay/cartridge/scripts/adyen3DVerification'), result, order, paymentInstrument,
 	adyenResponse  = session.custom.adyenResponse;
+
+	if(session.custom.orderNo && session.custom.paymentMethod) {
+		order = OrderMgr.getOrder(session.custom.orderNo);
+		paymentInstrument = order.getPaymentInstruments(session.custom.paymentMethod)[0];
```

Comment:
also here - but if you use it more than once maybe you can create a getter function for it where you handle if the instrument does not exits and then you are done everywhere :)